x-pack/plugins/security_solution/public/management/services/policies/hooks.ts

```diff
@@ -63,3 +84,199 @@ export function useGetEndpointSecurityPackage({
     customQueryOptions
   );
 }
+
+const DEFAULT_BROWSER_FIELDS = {};
+const DEFAULT_INDEX_PATTERNS = { fields: [], title: '' };
+
+interface FetchIndexReturn {
+  browserFields: BrowserFields;
+  indexes: string[];
+  indexExists: boolean;
+  indexPatterns: DataViewBase;
+}
+
+type DangerCastForBrowserFieldsMutation = Record<
+  string,
+  Omit<BrowserField, 'fields'> & { fields: Record<string, BrowserField> }
+>;
+interface DataViewInfo {
+  browserFields: DangerCastForBrowserFieldsMutation;
+  indexFields: FieldSpec[];
+}
+
+/**
+ * HOT Code path where the fields can be 16087 in length or larger. This is
+ * VERY mutatious on purpose to improve the performance of the transform.
+ */
+const getDataViewStateFromIndexFields = memoizeOne(
+  (_title: string, fields: IndexField[], _includeUnmapped: boolean = false): DataViewInfo => {
+    // Adds two dangerous casts to allow for mutations within this function
+    type DangerCastForMutation = Record<string, {}>;
+
+    return fields.reduce<DataViewInfo>(
+      (acc, field) => {
+        // mutate browserFields
+        if (acc.browserFields[field.category] == null) {
+          (acc.browserFields as DangerCastForMutation)[field.category] = {};
+        }
+        if (acc.browserFields[field.category].fields == null) {
+          acc.browserFields[field.category].fields = {};
+        }
+        acc.browserFields[field.category].fields[field.name] = field as unknown as BrowserField;
+
+        // mutate indexFields
+        acc.indexFields.push(
+          pick(['name', 'searchable', 'type', 'aggregatable', 'esTypes', 'subType'], field)
+        );
+
+        return acc;
+      },
+      {
+        browserFields: {},
+        indexFields: [],
+      }
+    );
+  },
+  (newArgs, lastArgs) =>
+    newArgs[0] === lastArgs[0] &&
+    newArgs[1].length === lastArgs[1].length &&
+    newArgs[2] === lastArgs[2]
+);
+
+export const getIndexFields = memoizeOne(
+  (title: string, fields: IndexField[], _includeUnmapped: boolean = false): DataViewBase =>
+    fields && fields.length > 0
+      ? {
+          fields: fields.map((field) =>
+            pick(
+              [
+                'name',
+                'searchable',
+                'type',
+                'aggregatable',
+                'esTypes',
+                'subType',
+                'conflictDescriptions',
+              ],
+              field
+            )
+          ),
+          title,
+        }
+      : { fields: [], title },
+  (newArgs, lastArgs) =>
+    newArgs[0] === lastArgs[0] &&
+    newArgs[1].length === lastArgs[1].length &&
+    newArgs[2] === lastArgs[2]
+);
+
+export function useFetchEventFiltersFields(
```

Comment:
Is this a copy of `useFetchIndex` function but using the ` ENDPOINT_FIELDS_SEARCH_STRATEGY` search strategy?
I would put this in another folder out of `policies` one since it's not related to policies.
Thanks for adding a fix for this!

Reply:
That would work, but if we want to reuse it for other use cases, I would put it in `x-pack/plugins/security_solution/public/management/services/search_strategy/hooks.ts` or in an upper level where we have other hooks.

Reply:
Looks like this is where most of the hooks are: `x-pack/plugins/security_solution/public/management/hooks`